### PR TITLE
Image Cache

### PR DESCRIPTION
**Pull request by amywieliczka**
Addresses #1109 

Stores thumbnail and media components on s3 at a configured `CONTENT_COMPONENT_CACHE` set to `s3://<bucket>/<prefix>/` at the key: `<collection_id>/<request.url>/<component_type: media|thumbnail>/<ETag>/<Last-Modified>`

Where request.url, etag, and last-modified have all been run through `urllib.parse.quote_plus` to encode them. 

Thumbnail and Media Components are defined as: 

```python
thumbnail_component = {
    'mimetype': 'image/jpeg',
    'path': content_s3_filepath,                # s3 filepath to generated thumbnail
    'dimensions': dimensions,                   # dimensions of generated thumbnail
    'component_content_harvest_metadata': {
        'md5': source_component['md5'],                                 # md5 of response content from mapped thumbnail source url
        'src_content-type': source_component['Content-Type'],           # content type response header from mapped thumbnail source url,
        'mapped_mimetype': mapped_thumbnail_source.get('mimetype'),     # mapped thumbnail source mimetype
        'src_size': source_component['size'],                           # size in bytes of response content from mapped thumbnail source url
        'date_content_component_created': datetime.now().isoformat()
    }
}

media_component = {
    'mimetype': mimetype,                                           # mimetype of generated media
    'path': content_s3_filepath,                                    # s3 path to generated media
    'format': NUXEO_MEDIA_TYPE_MAP.get(mapped_nuxeotype or ''),     # format, used by UI to determine display
    'component_content_harvest_metadata': {
        'md5': source_component['md5'],                             # md5 hash of source file
        'src_content-type': source_component['Content-Type'],       # content type response header from source file
        'mapped_mimetype': mapped_mimetype,                         # mimetype from mapper
        'src_size': source_component['size'],                       # size in bytes of source file
        'date_content_component_created': datetime.now().isoformat()
    }
}
```

### CODE REVIEW

File: content_harvester/s3_cache.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            raise e
+    
+    def __getitem__(self, key):
+        """
+        Get the value for the key using bracket notation: S3Cache[key]
+        Raise all exceptions with added key context
+        """
+        try:
+            response = self.s3.get_object(
+                Bucket=self.bucket_name, Key=self.prefix+key)
```
Comment (bibliotechy) [inline content_harvester/s3_cache.py:32]: Should we wrap this in an f-string just to be safe?

**Review by bibliotechy — APPROVED** [review]
s3 implementation looks good. One small non-blocker comment

File: content_harvester/s3_cache.py (hunk 1 above)
Comment (amywieliczka) [reply content_harvester/s3_cache.py]: Done!

**Review by barbarahui — APPROVED** [review]
Neat!!

**Pull request merged**